Kitodo/src/test/java/test/SimpleLoginST.java

```diff
@@ -0,0 +1,411 @@
+                    BufferedOutputStream bufferedOutputStream = new BufferedOutputStream(fileOutputStream, BUFFER_SIZE);
+
+                    while ((currentByte = bufferedInputStream.read(data, 0, BUFFER_SIZE)) != -1) {
+                        bufferedOutputStream.write(data, 0, currentByte);
+                    }
+                    bufferedOutputStream.flush();
+                    bufferedOutputStream.close();
+                    bufferedInputStream.close();
+                }
+            }
+        } catch (Exception e) {
+            logger.error(e.getMessage());
+        }
+    }
+
+    /**
+     * Unpack the archive to specified directory.
+     *
+     * @param file
+     *            The tar or tar.gz file.
+     * @param outputDir
+     *            The destination directory.
+     * @param isGZipped
+     *            True if the file is gzipped.
+     */
+    private static void extractTarFileToFolder(File file, File outputDir, boolean isGZipped) {
+        FileInputStream fileInputStream = null;
+        TarInputStream tarArchiveInputStream = null;
+        try {
+            fileInputStream = new FileInputStream(file);
+            tarArchiveInputStream = (isGZipped) ? new TarInputStream(new GZIPInputStream(fileInputStream, BUFFER_SIZE))
+                    : new TarInputStream(new BufferedInputStream(fileInputStream, BUFFER_SIZE));
+            unTar(tarArchiveInputStream, outputDir);
+        } catch (IOException e) {
+            logger.error(e.getMessage());
+        } finally {
+            if (tarArchiveInputStream != null) {
+                try {
+                    tarArchiveInputStream.close();
+                } catch (IOException e) {
+                    logger.error(e.getMessage());
+                }
+            } else if (fileInputStream != null) {
+                try {
+                    fileInputStream.close();
+                } catch (IOException e) {
+                    logger.error(e.getMessage());
+                }
+            }
+        }
+    }
+
+    /**
+     * Unpack data from the stream to specified directory.
+     *
+     * @param tarInputStream
+     *            Stream with tar data.
+     * @param outputDir
+     *            The destination directory.
+     */
+    private static void unTar(TarInputStream tarInputStream, File outputDir) {
+        try {
+            TarEntry tarEntry;
+            while ((tarEntry = tarInputStream.getNextEntry()) != null) {
+                final File file = new File(outputDir, tarEntry.getName());
+                if (tarEntry.isDirectory()) {
+                    if (!file.exists()) {
+                        if (!file.mkdirs()) {
+                            logger.error(file + " failure to create directory");
+                        }
+                    }
+                } else {
+                    BufferedOutputStream out = new BufferedOutputStream(new FileOutputStream(file));
```

Comment:
Same here for `out`: use the [try-with-resources-statement](https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html).